src/container.c

```diff
@@ -424,23 +424,24 @@ static int container_binding_file(char *src, char *dest)
 
 	if (stat(src, &st) < 0) {
 		if (errno == ENOENT) {
-			fprintf(stdout, "no dns configured\n");
+			fprintf(stdout, "can not find %s\n", src);
 			return 0;
 		}
 
-		fprintf(stderr, "stat %s failed", src);
+		fprintf(stderr, "stat %s failed: %s\n", src, strerror(errno));
 		return -1;
 	}
 
+	// root filesystem may be read only
 	fd = open(dest, O_CREAT| O_WRONLY, 0644);
```

Comment:
It is not about EEXIST but EROFS. Since we open with `O_CREAT|O_WRONLY`, it fails on a RO root fs. But if we stat() the dest file and skip opening if it exists, we can still bind mount to it even on a RO root fs.